src/compiler/program.ts

```diff
@@ -1703,7 +1703,11 @@ namespace ts {
             fail?: (diagnostic: DiagnosticMessage, ...argument: string[]) => void,
             refFile?: SourceFile): SourceFile | undefined {
 
-            if (hasExtension(fileName)) {
+            // Don't process falsy fileNames

```

Comment:
how about doing this check on the call to `getDefaultLibraryFileName` instead. it is making me a bit nervous to have it in this general utility function, just to handle the return of `getDefaultLibraryFileName`.

Reply:
Making you nervous is not a good sign.

You mean here: https://github.com/alan-agius4/TypeScript/blob/a27cd8f9a2e83d4b6fbf4e77c5041d94b01d86f6/src/compiler/program.ts#L611

Change it to: `if (!skipDefaultLib && defaultLibraryPath) `